laspy/copc.py

```diff
@@ -693,11 +695,12 @@ def open(
         .. versionadded:: 2.4
             The ``decompression_selection`` parameter.
         """
-        uri = str(uri)
-        if uri.startswith("http"):
-            source = HttpRangeStream(uri)
-        else:
-            source = open(uri, mode="rb")
+        if isinstance(source, str) or isinstance(source, os.PathLike):
+            source = str(source)
+            if source.startswith("http"):
```

Comment:
https, s3, gcs, and other remote protocols too?

Reply:
`startswith('https')` will also catch https endpoints.

laspy does not support s3/gcs protocols by itself, but the user can provide the source as a fileobject and things will work (as long as said file object has the required methods)

```python
import laspy
import s3fs

fs = s3fs.S3FileSystem()
with fs.open("s3://bucket/object.laz") as fh:
    with laspy.open(fh) as las_fh:
```

Reply:
I haven't done any serious point reads with the fsspec libraries, but typically use this to quickly read headers.